doc/developer/platform/ux.md

```diff
@@ -1,20 +1,726 @@
+access to billing and enabling new regions) and the ability to invite new users
+to the account, while account members do not. The first user in an account is
+always an account owner. Account owners are effectively superusers of the region
+API, and are not subject to any [per-region roles](#role) when accessing the
+region API.
+
+### App-specific password
+
+An *app-specific password* is an alternate encoding of a personal API token
+for authenticating with the PostgreSQL protocol. Users should create an
+app-specific password for each external service that will connect to Materialize
+Platform via the PostgreSQL protocol.
+
+Rationale: the PostgreSQL protocol does not have separate fields for an API
+client ID and API secret key, so both values need to be packed into the password
+field. Naively packing the UUIDs is unwieldy, however:
+
+```
+893677c2-a74c-4ecb-804d-3b7fa0c61fdc:68104da2-3417-442e-a614-f078dac470be
+```
+
+But since UUIDs are just 16 bytes each, we can pack them into a much shorter
+base64-encoded string:
+
+```py
+>>> base64.urlsafe_b64encode(client_id.bytes + secret_key.bytes).decode().rstrip('=')
+iTZ3wqdMTsuATTt_oMYf3GgQTaI0F0QuphTweNrEcL4
+```
+
+When encoded in this form, we call the value an app-specific password rather
+than an API key.
+
+In the web UI, we will need to build a custom interface for managing personal
+API tokens that shows the app-specific password form alongside the client ID
+and secret.
+
+### Billing
+
+Materialize Platform will bill monthly based on usage. Accounts will be billed
+in terms of compute units and storage units. A strawman proposal: compute units
+are machine-hours and storage units are gigabyte-months.
+
+We expect metering the compute usage of clusters and the storage usage of
+sources and tables to be straightforward. What is not clear is how to bill for
+the overhead of other system components.
+
+### Region
+
+When an account is created, it does not have any regions. The user's first step
+after registering for a new account is to choose one region to activate.
+
+Additional regions can be enabled via a "+ Enable new region" button in the
+region selector or somesuch in the web UI, or via the [global API](#global-api).
+
+To prevent disaster, regions cannot be deactivated. Users who wish to deactivate
+a region can contact support.
+
+### Cluster
+
+Clusters are created within a region via a new `CREATE CLUSTER` SQL statement:
+
+```sql
+CREATE CLUSTER <name> SIZE {xs|s|m|l|xl} [, REPLICATION FACTOR <n>];
+```
+
+Creating a cluster allocates persistent compute resources (e.g., EC2 instances)
+and begins billing the account for those resources. The SQL statement returns
+successfully as soon as the request is received.
+
+Specifying a replication factor greater than one will run multiple identical
+replicas of the cluster for redundancy. If one replica fails, requests can
+be routed to the other replica(s) until the failed replica is brought back
+online and brought up to date.
```

Comment:
idk if this matters, but I would guess active replication is more likely, and it is less "if one fails we can then use the other" and more "commands are sent to both, so that if either is unavailable ..."

Reply:
Oops, yeah, that's confusingly phrased. It does not intend to imply we'll do failover instead of active–active, and I think taken extremely literally it does not, but I will fix.